index.js

```diff
@@ -124,10 +125,66 @@ function parseFileExtension(opts) {
   return opts;
 }
 
+// mongodb-linux-x86_64-rhel62-3.6.2.tgz
+//
+// Red Hat Enterprise Linux Server release 7.2 (Maipo)
+function parseRhel(data) {
+  if (data.match('7.1')) return 'rhel71';
+  if (data.match('6.0')) return 'rhel60';
+  return new Error('Could not determine RHEL version');
+}
+
+//  mongodb-linux-x86_64-suse12-3.6.2.tgz
+//
+//  SUSE Linux Enterprise Server 11 (x86_64)
+//  VERSION = 11
+//  PATCHLEVEL = 4
+function parseSuse(data) {
+  if (data.match('Server 11')) return 'suse11';
+  if (data.match('Server 12')) return 'suse12';
+  return new Error('Could not determine SUSE version');
+}
+
+// mongodb-linux-x86_64-ubuntu1204-3.6.2.tgz
+//
+// DISTRIB_ID=Ubuntu
+// DISTRIB_RELEASE=16.04
+// DISTRIB_CODENAME=xenial
+// DISTRIB_DESCRIPTION="Ubuntu 16.04 LTS"
+function parseUbuntu(data) {
+  if (data.match('16.04')) return 'ubuntu1604';
+  if (data.match('14.04')) return 'ubuntu1404';
+  if (data.match('12.04')) return 'ubuntu1204';
+  return new Error('Could not determine Ubuntu version');
+}
+
+function getLinuxDistro() {
+  var distros = [
+    ['/etc/redhat-release', parseRhel],
+    ['/etc/SuSE-release', parseSuse],
+    ['/etc/lsb-release', parseUbuntu]
+  ];
+
+  var formattedDistro = distros.reduce(function(result, distro) {
+    if (result) return result;
+    try {
+      return distro[1](fs.readFileSync(distro[0]).toString()); // eslint-disable-line
+    } catch (error) {
+      // Catching file reading error
+      return new Error(error);
+    }
+  }, null);
+  if (formattedDistro instanceof Error) {
+    // Catching parsing version error
+    throw formattedDistro;
```

Comment:
@mbroadst @daprahamian This is now switched over to `reduce` but to be honest I think it's messy to twist reduce to work for us in this case. I want to double check that we're ok throwing here?